core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/AbstractContractNegotiationManager.java

```diff
@@ -165,8 +165,7 @@ public T build() {
         }
     }
 
-    protected void update(ContractNegotiation negotiation, Consumer<ContractNegotiationListener> observe) {
-        observable.invokeForEach(observe);
+    protected void update(ContractNegotiation negotiation) {
```

Comment:
At this point either inline the method or pass the post-action consumer to be called after the update when it's needed

Reply:
I made the function `final`. Hope thats the correct way, you meant it.

Reply:
@janpmeyer no, I didn't mean that, making it `final` doesn't bring any value in this case.
What I was saying is that having a method with just one line probably is a sign that the method can be inlined.

Reply:
Should be fixed now